src/gui/torrentcategorydialog.cpp

```diff
@@ -128,12 +133,58 @@ void TorrentCategoryDialog::setCategoryName(const QString &categoryName)
     m_ui->textCategoryName->setText(categoryName);
 }
 
-QString TorrentCategoryDialog::savePath() const
+BitTorrent::CategoryOptions TorrentCategoryDialog::categoryOptions() const
+{
+    BitTorrent::CategoryOptions categoryOptions;
+    categoryOptions.savePath = m_ui->comboSavePath->selectedPath();
+    if (m_ui->comboUseDownloadPath->currentIndex() == 1)
+        categoryOptions.downloadPath = {true, m_ui->comboDownloadPath->selectedPath()};
+    else if (m_ui->comboUseDownloadPath->currentIndex() == 2)
+        categoryOptions.downloadPath = {false};
+
+    return categoryOptions;
+}
+
+void TorrentCategoryDialog::setCategoryOptions(const BitTorrent::CategoryOptions &categoryOptions)
+{
+    m_ui->comboSavePath->setSelectedPath(categoryOptions.savePath);
+    if (categoryOptions.downloadPath)
+    {
+        m_ui->comboUseDownloadPath->setCurrentIndex(categoryOptions.downloadPath->enabled ? 1 : 2);
+        m_ui->comboDownloadPath->setSelectedPath(categoryOptions.downloadPath->enabled ? categoryOptions.downloadPath->path : QString());
+    }
+    else
+    {
+        m_ui->comboUseDownloadPath->setCurrentIndex(0);
+        m_ui->comboDownloadPath->setSelectedPath({});
+    }
+}
+
+void TorrentCategoryDialog::categoryNameChanged(const QString &categoryName)
 {
-    return m_ui->comboSavePath->selectedPath();
+    using namespace Utils::Fs;
+
+    const QString categoryPath = toValidFileSystemName(categoryName, true);
+    const auto *btSession = BitTorrent::Session::instance();
+    m_ui->comboSavePath->setPlaceholder(resolvePath(categoryPath, btSession->savePath()));
+
+    if (m_ui->comboUseDownloadPath->currentIndex() != 2)
+        m_ui->comboDownloadPath->setPlaceholder(resolvePath(categoryPath, btSession->downloadPath()));
+
+    m_ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(!categoryName.isEmpty());
 }
 
-void TorrentCategoryDialog::setSavePath(const QString &savePath)
+void TorrentCategoryDialog::useDownloadPathChanged(const int index)
 {
-    m_ui->comboSavePath->setSelectedPath(savePath);
+    if (const QString selectedPath = m_ui->comboDownloadPath->selectedPath(); !selectedPath.isEmpty())
+        m_lastEnteredDownloadPath = selectedPath;
+
+    m_ui->comboDownloadPath->setEnabled(index == 1);
```

Comment:
Disable the label too:
`m_ui->labelDownloadPath->setEnabled(index == 1);`

These 2 commands need to be called in constructor too or somewhere else on dialog creation. 
And while we're in this method, the "Path" should be empty when the global temp path setting is disabled. I don't know about the rest of the method, i only noticed these.

Reply:
Should be fixed now.